### PR TITLE
Unnecessary code removal

### PR DESCRIPTION
**Pull request by xtonik**
### SUMMARY
Removed various unnecessary code.

### Automated Checks

- [x] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [x] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

**Review by richardstartin — DISMISSED** [review]
Please do not remove the reference check, it is an optimisation

File: RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java [hunk 1]
```diff
@@ -445,10 +445,7 @@ private static int workShyAndCardinality(RoaringBitmap... bitmaps) {
           continue;
         }
         Container container = bitmap.highLowContainer.getContainerAtIndex(index);
-        Container and = tmp.iand(container);
```
Comment (richardstartin) [inline RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java:448]: This is an optimisation

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java [hunk 2]
```diff
@@ -560,10 +556,7 @@ static MutableRoaringBitmap workShyAnd(long[] aggregationBuffer,
       Arrays.fill(words, -1L);
       MappeableContainer tmp = new MappeableBitmapContainer(LongBuffer.wrap(words), -1);
       for (MappeableContainer container : slice) {
-        MappeableContainer and = tmp.iand(container);
```
Comment (richardstartin) [inline RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java:563]: This is an optimisation

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java [hunk 3]
```diff
@@ -79,11 +79,7 @@ public static MutableRoaringBitmap and(Iterator<? extends ImmutableRoaringBitmap
   public static MutableRoaringBitmap and(long[] aggregationBuffer,
                                          Iterator<? extends ImmutableRoaringBitmap> bitmaps) {
     if (bitmaps.hasNext()) {
-      try {
-        return workShyAnd(aggregationBuffer, bitmaps);
-      } finally {
-        Arrays.fill(aggregationBuffer, 0L);
-      }
+      return workShyAnd(aggregationBuffer, bitmaps);
```
Comment (richardstartin) [inline RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java:82]: Behaviour change

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java (hunk 2 above)
Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java:563]: @richardstartin Same question ... can you explain (briefly) the optimization?

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java (hunk 3 above)
Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java:82]: @richardstartin What triggers the exception in this instance?

File: RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java (hunk 1 above)
Comment (richardstartin) [reply RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java:448]: this increases throughput when G1 is the active GC by avoiding its write barrier, which can be surprisingly expensive in tight loops

Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java:448]: Ok. Please see https://github.com/RoaringBitmap/RoaringBitmap/pull/672

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java (hunk 2 above)
Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java:563]: Answered above.

File: RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java (hunk 1 above)
Comment (xtonik) [reply RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java]: Sorry, I did not realize that. I rolled back all related changes.

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java (hunk 3 above)
Comment (xtonik) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java]: You are right, above that the use without clearing buffer does not make sense or it is better to be done at once.

What should be changed I think is to get of rid `finally` section completely as some unexpected exception can occur and then it is ignored or it is also related to GC write barrier? Above that in the case with provided iterator the method returns empty bit map if some exception occurs, which I mean it is not expected behavior. Imagine the use case: roaring bit map used as a ip blacklist and we want match ips from all incoming request with blacklisted ones. The result is no one is on the blacklist when used like `ips.and(buffer, blacklist)`.

What's next is that when the sibling method `RoaringBitmap and(RoaringBitmap... bitmaps) `without provided buffer is called, buffer is uselessly cleared at the end of work even that it is not used at all after, may be it is skipped by JVM optimization, don't know.

Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java]: @richardstartin Can you help us understand this code?

Comment (lemire) [reply RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java]: > as some unexpected exception can occur and then it is ignored

Finally would not ignore the exception. This code still throws an exception.

https://stackoverflow.com/questions/2614473/does-it-make-sense-to-do-try-finally-without-catch

**Comment by lemire** [comment]
@xtonik Please split this PR into smaller PRs. Right now I cannot merge this because there are changes that I cannot totally understand.

**Comment by xtonik** [comment]
Sorry for this mesh, PR is now split to #678 and #679.

**Pull request closed without merging**